pkg/storaged/partitions/partition.jsx

```diff
@@ -118,6 +122,62 @@ const PartitionCard = ({ card, block, unused_space_warning, resize_info }) => {
         return grow_dialog(client, block_part, resize_info, true);
     }
 
+    const gpt_type_names = {
+        "21686148-6449-6E6F-744E-656564454649": _("BIOS boot partition"),
+        "C12A7328-F81F-11D2-BA4B-00A0C93EC93B": _("EFI system partition"),
+        "9E1A2D38-C612-4316-AA26-8B49521E5A8B": _("PowerPC PReP boot partition"),
+        "0657FD6D-A4AB-43C4-84E5-0933C84B4F4F": _("Linux swap space"),
+        "0FC63DAF-8483-4772-8E79-3D69D8477DE4": _("Linux filesystem data"),
+        "E6D6D379-F507-44C2-A23C-238F2A3DF928": _("Logical Volume Manager partition"),
+    };
+
+    const mbr_type_names = {
+        "0XEF": _("EFI system partition"),
+        "0X82": _("Linux swap space"),
+        "0X83": _("Linux filesystem data"),
+        "0X8E": _("Logical Volume Manager partition"),
+    };
+
+    function partition_type_name(type) {
+        const t = type.toUpperCase();
+        return gpt_type_names[t] || mbr_type_names[t] || type;
+    }
+
+    function set_type_dialog() {
+        const block_ptable = client.blocks_ptable[block_part.Table];
+        if (!block_ptable)
+            return;
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19800-20240102-135227-86122ba7-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)